subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/TypeValidationProblemRenderer.java

```diff
@@ -84,15 +85,24 @@ public static String convertToSingleLine(String message) {
     private static String introductionFor(TypeValidationProblemLocation location) {
         StringBuilder builder = new StringBuilder();
         Class<?> rootType = location.getType().orElse(null);
+        PluginId pluginId = location.getPlugin().orElse(null);
         if (rootType != null) {
-            builder.append("Type '");
+            if (pluginId != null) {
+                builder.append("In plugin '").append(pluginId).append("' type '");
+            } else {
+                builder.append("Type '");
```

Comment:
I think we should append the fully qualified name of the type here when we don't know the plugin id. WDYT?

Reply:
I actually didn't realize we only display the short name. Would it actually make sense to always display the FQN?

Reply:
It does. Actually it makes the problem of `DefaultTask` even worse, since we don't detect an associated plugin, we now show the fqn for it.

Reply:
done